### PR TITLE
run the db backup via restic, not as a separate cron job

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: puppet/modules/profiles/manifests/redmine.pp [hunk 1]
```diff
@@ -5,11 +5,14 @@
   # TODO include redmine
   $backup_path = '/var/lib/redmine'
 
-  include profiles::backup::sender
+  class {'profiles::backup::sender':
```
Comment (ekohl) [inline puppet/modules/profiles/manifests/redmine.pp:8]: We have `restic::user_homedir`:
https://github.com/theforeman/foreman-infra/blob/23157ca170975c0be21046473e02be6236cf64af/puppet/modules/profiles/manifests/backup/sender.pp#L19

I wonder if we set that one correctly. Makes me question if we want the process to also have access to the the SSH key. If the service is compromised they can also easily destroy all backups.

Thinking out loud: could we use https://systemd.io/CREDENTIALS/ to let systemd provide the key in $CREDENTIALS_DIRECTORY? You can override some locations with env vars: https://restic.readthedocs.io/en/stable/040_backup.html#environment-variables

https://restic.readthedocs.io/en/stable/030_preparing_a_new_repo.html#sftp is not very clear in an easy way to override the key location via env vars.

File: puppet/modules/profiles/manifests/redmine.pp [hunk 2]
```diff
@@ -5,11 +5,14 @@
...
+    username => $redmine::username,
+  }
 
   restic::repository { 'redmine':
-    backup_cap_dac_read_search => true,
-    backup_path                => $backup_path,
-    backup_flags               => ['--exclude', "${backup_path}/git"],
+    backup_path    => $backup_path,
+    backup_flags   => ['--exclude', "${backup_path}/git"],
+    backup_pre_cmd => "pg_dump redmine | gzip > ${backup_path}/redmine.backup.sql.gz",
```
[the remark this answers is not in the document]
Comment (evgeni) [reply puppet/modules/profiles/manifests/redmine.pp:15]: I think modern restic does. Will investigate

[the remark this answers is not in the document]
Comment (evgeni) [reply puppet/modules/profiles/manifests/redmine.pp:15]: https://restic.readthedocs.io/en/latest/047_tuning_backup_parameters.html#compression - yes it does, if you have repo version 2, which you do if you created the repo with restic 0.14+

File: puppet/modules/profiles/manifests/redmine.pp (hunk 1 above)
Comment (evgeni) [reply puppet/modules/profiles/manifests/redmine.pp:8]: my private ssh config is sprinkled with `IdentityFile ~/.ssh/backup_key`, so I guess the answer is "no"

**Comment by evgeni** [comment]
grant alternative: https://github.com/theforeman/foreman-infra/pull/1893

**Comment by evgeni** [comment]
sudo alternative: https://github.com/theforeman/foreman-infra/pull/1894

**Comment by evgeni** [comment]
closed in favor of #1893

**Pull request closed without merging**